src/lib/commands.js

```diff
@@ -0,0 +1,274 @@
+	encodeConsentData
+} from "./cookie/cookie";
+import {
+	CMP_VERSION,
+	CMP_ID,
+	COOKIE_VERSION
+} from "./init";
+import {SECTION_DETAILS, SECTION_VENDORS} from "./store";
+import log from "./log";
+
+const arrayFrom = require('core-js/library/fn/array/from');
+const COOKIE_VERSION_V1 = 1;
+
+const createCommands = (store) => {
+
+	/**
+	 * Converts vector to consent object
+	 */
+	const vectorToObject = (set) => {
+		let obj = {};
+
+		set.forEach((hasConsent, id) => {
+			obj[id] = hasConsent;
+		});
+
+		return obj;
+	};
+
+	 /**
+	 * Build consent fields object v1 from data that has already been persisted.
+	 */
+	const getConsentFieldsObject = () => {
+		const {
+			persistedConsentData
+		} = store;
+
+		const {
+			created,
+			lastUpdated,
+			cmpId_: cmpId = store.tcModel.cmpId,
+			cmpVersion_: cmpVersion = 1,
+			consentScreen_: consentScreen = store.tcModel.consentScreen,
+			consentLanguage_: consentLanguage = store.tcModel.consentLanguage,
+			vendorListVersion_: vendorListVersion,
+		} = persistedConsentData;
+
+		return {
+			cmpId,
+			cmpVersion,
+			consentLanguage,
+			consentScreen,
+			created,
+			vendorListVersion,
+			lastUpdated,
+		};
+	};
+
+	const generateConsentStringV1 = (data = {}, callback) => {
+		const {
+			persistedConsentData,
+		} = store;
+
+		const {
+			created,
+			lastUpdated,
+			consentLanguage_: consentLanguage,
+			consentScreen_: consentScreen,
+			version_: version = COOKIE_VERSION_V1,
+			cmpId_: cmpId = CMP_ID
+		} = persistedConsentData;
+
+		const {
+			selectedVendorIds = [],
+			selectedPurposeIds = [],
+			maxVendorId,
+			cookieVersion,
+			vendorListVersion,
+			cmpVersion
+		} = data;
+
+		const consentData = {
+			cmpId,
+			maxVendorId,
+			cookieVersion,
+			cmpVersion,
+			vendorListVersion,
+			created,
+			lastUpdated,
+			consentLanguage,
+			consentScreen,
+			version,
+			selectedVendorIds: new Set(arrayFrom(selectedVendorIds)),
+			selectedPurposeIds: new Set(arrayFrom(selectedPurposeIds))
+		};
+
+		const valid = (data = {}) => [
+			'cmpId', 'cmpVersion', 'consentLanguage', 'consentScreen', 'cookieVersion', 'created', 'lastUpdated',
+			'maxVendorId', 'selectedPurposeIds', 'selectedVendorIds', 'vendorListVersion'
+		].every(prop => data.hasOwnProperty(prop));
+
+		callback(valid(consentData) ? encodeVendorConsentData(consentData) : undefined);
+	};
+
+	/**
+	 * Generates v1.1 compatible TC String
+	 */
+	const getConsentFieldsV1 = (callback, params) => {
+		const data = getConsentFieldsObject();
+
+		generateConsentStringV1({
+			...data,
+			...params
+		}, (metadata) => {
+			const consent = {
+				metadata,
+				gdprApplies: config.gdprApplies,
+				hasGlobalScope: config.storeConsentGlobally,
+				...data,
+				vendorListVersion: params.vendorListVersion
+			};
+
+			callback(consent, true);
+		});
+	};
+
+	/**
+	 * Generates consent object based on input params: consents and vendor list
+	 */
+	const getConsentObject = (callback, consents, vendorList) => {
```

Comment:
TCFv2 standard [states](https://github.com/InteractiveAdvertisingBureau/GDPR-Transparency-and-Consent-Framework/blob/master/TCFv2/IAB%20Tech%20Lab%20-%20CMP%20API%20v2.md#how-does-the-cmp-provide-the-api) that every command should have at most one parameter.

Reply:
It seems that they decided to support it - https://github.com/InteractiveAdvertisingBureau/iabtcf-es/issues/82